Mathlib/Data/Fin/SuccPred.lean

```diff
@@ -22,50 +22,76 @@ namespace Fin
 instance : ∀ {n : ℕ}, SuccOrder (Fin n)
   | 0 => by constructor <;> intro a <;> exact elim0 a
   | n + 1 =>
-    SuccOrder.ofCore (fun i => if i < Fin.last n then i + 1 else i)
-      (by
-        intro a ha b
-        rw [isMax_iff_eq_top, eq_top_iff, not_le, top_eq_last] at ha
+    SuccOrder.ofCore
+      (fun i ↦ if hi : i = Fin.last n
+        then Fin.last n else i.succ.castPred (by simpa [← Fin.succ_last]))
```

Comment:
Can we use `Fin.lastCases` here (and `Fin.cases` below) instead?

Reply:
Thanks very much for the suggestion! The code looks much nicer now.